pkg/controller/nmstate/nmstate_controller.go

```diff
@@ -145,11 +147,35 @@ func (r *ReconcileNMState) applyRBAC(instance *nmstatev1beta1.NMState) error {
 
 func (r *ReconcileNMState) applyHandler(instance *nmstatev1beta1.NMState) error {
 	data := render.MakeRenderData()
+	// Register toYaml function
+	data.Funcs["toYaml"] = helper.ToYaml
+
+	// Prepare defaults
+	masterExistsNoScheduleToleration := corev1.Toleration{
+		Key:      "node-role.kubernetes.io/master",
+		Operator: corev1.TolerationOpExists,
+		Value:    string(corev1.TaintEffectNoSchedule),
+	}
+	handlerNodeSelector := instance.Spec.NodeSelector
+	handlerNodeSelector["beta.kubernetes.io/arch"] = "amd64"
+
 	data.Data["HandlerNamespace"] = os.Getenv("HANDLER_NAMESPACE")
 	data.Data["HandlerImage"] = os.Getenv("HANDLER_IMAGE")
 	data.Data["HandlerPullPolicy"] = os.Getenv("HANDLER_IMAGE_PULL_POLICY")
 	data.Data["HandlerPrefix"] = os.Getenv("HANDLER_PREFIX")
-	data.Data["HandlerNodeSelector"] = instance.Spec.NodeSelector
+	data.Data["WebhookNodeSelector"] = map[string]string{
+		"beta.kubernetes.io/arch":        "amd64",
+		"node-role.kubernetes.io/master": "",
```

Comment:
Can we move this to defaults section at this function ?

Reply:
Done